test/cppsuite/tests/cache_resize.cpp

```diff
@@ -0,0 +1,223 @@
+    {
+        const uint64_t collection_count = tc->db.get_collection_count();
+        testutil_assert(collection_count > 0);
+        collection &coll = tc->db.get_collection(collection_count - 1);
+        scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+
+        while (tc->running()) {
+            tc->sleep();
+
+            /* Insert the current cache size value using a random key. */
+            const std::string key =
+              random_generator::instance().generate_pseudo_random_string(tc->key_size);
+            const uint64_t cache_size =
+              ((WT_CONNECTION_IMPL *)connection_manager::instance().get_connection())->cache_size;
+            /* Take into account the value size given in the test configuration file. */
+            const std::string value = tc->left_padding_to_string(cache_size, tc->value_size);
+
+            tc->transaction.try_begin();
+            if (!tc->insert(cursor, coll.id, key, value)) {
+                tc->transaction.rollback();
+            } else if (tc->transaction.can_commit()) {
+                /*
+                 * The transaction can fit in the current cache size and is ready to be committed.
+                 * This means the tracking table will contain a new record to represent this
+                 * transaction which will be used during the validation stage.
+                 */
+                testutil_assert(tc->transaction.commit());
+            }
+        }
+
+        /* Make sure the last transaction is rolled back now the work is finished. */
+        if (tc->transaction.active())
+            tc->transaction.rollback();
+    }
+
+    void
+    validate(const std::string &operation_table_name, const std::string &,
+      const std::vector<uint64_t> &) override final
+    {
+        bool first_record = false;
+        int ret;
+        uint64_t cache_size, num_records = 0, prev_txn_id;
+        const uint64_t cache_size_500mb = 500000000;
+
+        /* Open a cursor on the tracking table to read it. */
+        scoped_session session = connection_manager::instance().create_session();
+        scoped_cursor cursor = session.open_scoped_cursor(operation_table_name);
+
+        /*
+         * Parse the tracking table. Each operation is tracked and each transaction is made of
+         * multiple operations, hence we expect multiple records for each transaction. We only need
+         * to verify that the cache size was big enough when the transaction was committed, which
+         * means at the last operation.
+         */
+        while ((ret = cursor->next(cursor.get())) == 0) {
+
+            uint64_t tracked_ts, tracked_txn_id;
+            int tracked_op_type;
+            const char *tracked_cache_size;
+
+            testutil_check(cursor->get_key(cursor.get(), &tracked_ts, &tracked_txn_id));
+            testutil_check(cursor->get_value(cursor.get(), &tracked_op_type, &tracked_cache_size));
+
+            logger::log_msg(LOG_TRACE,
```

Comment:
Do cache resizes have a txn_id or can we only print this for insert operations?

Reply:
Yes, they do! See [here](https://github.com/wiredtiger/wiredtiger/pull/7902/files#diff-e12c1ae126af9008c48745f0d1603ef8ad756233627ccc445aa530a801c00b01R99-R105).